mm2src/common/log.rs

```diff
@@ -97,9 +96,9 @@ impl Gravity {
     /// `println!` is used for compatibility with unit test stdout capturing.
     #[cfg(not(target_arch = "wasm32"))]
     fn flush(&self) {
-        let mut tail = self.tail.spinlock(77).unwrap();
+        let logged_with_log_output = LOG_CALLBACK.lock().is_some();
+        let mut tail = self.tail.lock();
```

Comment:
Would it make a difference if this lock was moved inside the while loop just before using the `tail` variable?

Reply:
Although `PaMutex` is a spinlock based on atomic operations so it can be overhead.
Also `PaMutex` is designed as fair lock so sequential calls to `PaMutex::lock` in a loop can lead to a switch context.